xbmc/filesystem/FileCache.cpp

```diff
@@ -312,25 +340,30 @@ void CFileCache::Process()
       }
     }
 
-    const int64_t maxWrite = m_pCache->GetMaxWriteSize(m_chunkSize);
-    int64_t maxSourceRead = m_chunkSize;
+    const int64_t maxWrite = m_pCache->GetMaxWriteSize(requestSize);
+    // Make maxSourceRead a multiple of the chunk size, bounded by maxWrite
+    // If we are falling behind we will use larger and larger reads which
+    // over network file systems will hopefully increase throughput.
+    // Special case: m_chunkSize == 1 (non-buffered file) fits
+    // right into this calculation.
+    int64_t maxSourceRead = maxWrite - maxWrite % m_chunkSize;
     // Cap source read size by space available between current write position and EOF
     if (m_fileSize != 0)
       maxSourceRead = std::min(maxSourceRead, m_fileSize - m_writePos);
 
     /* Only read from source if there's enough write space in the cache
      * else we may keep disposing data and seeking back on (slow) source
      */
-    if (maxWrite < maxSourceRead)
+    if (maxSourceRead <= 0)
     {
       // Wait until sufficient cache write space is available
       m_pCache->m_space.Wait(5ms);
       continue;
     }
 
-    ssize_t iRead = 0;
-    if (maxSourceRead > 0)
-      iRead = m_source.Read(buffer.get(), maxSourceRead);
+    CLog::Log(LOGDEBUG, "CFileCache::{} Reading {} bytes from source. maxWrite: {}",
+              __FUNCTION__, maxSourceRead, maxWrite);
```

Comment:
```suggestion
              __func__, maxSourceRead, maxWrite);
```

Reply:
TODO: this is way too spammy, so either it should be component logging or removed before merge

Reply:
I see a potential need for it in the future, how do I go about making it component logging? I tried grepping a little bit but didn't quite figure out what I need to do wrt the UI list of components.

Reply:
May bad, component logging is only used for extra logging from external libraries.
Not quite sure how to deal with the spam here, maybe this can only be logged when we're about to run dry and skip it when all is good? Other thoughts?

Reply:
Something might be possible.

We have the m_writeRate that we might use to figure out if we're having trouble keeping up, perhaps.

Or only logging when the buffer is almost empty like you said.

Reply:
Hopefully it's less spammy now, at least for operation where there is some headroom.

With my large-file-over-VPN it still takes about a minute until it stops printing. Not a huge issue there as that is about one line per second, but it could be worse for non-SMB sources.

Tell me what you think!